nixos/modules/services/misc/nzbhydra2.nix

```diff
@@ -0,0 +1,79 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let cfg = config.services.nzbhydra2;
+
+in {
+  options = {
+    services.nzbhydra2 = {
+      enable = mkEnableOption "NZBHydra2";
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/nzbhydra2/.config/NZBHydra2";
+        description = "The directory where NZBHydra2 stores its data files.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description =
+          "Open ports in the firewall for the NZBHydra2 web interface.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "nzbhydra2";
+        description = "User account under which NZBHydra2 runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "nzbhydra2";
+        description = "Group under which NZBHydra2 runs.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nzbhydra2;
+        defaultText = "pkgs.nzbhydra2";
+        description = "NZBHydra2 package to use.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules =
+      [ "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -" ];
+
+    systemd.services.nzbhydra2 = {
+      description = "NZBHydra2";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
```

Comment:
From the upstream website:
> Bugs may/do exist. Don't run it on the machine where the nuclear launch codes are stored.

I'm always a fan of hardening `systemd` services, at least minimally, so maybe we can add a few flags to this unit.

If you have no idea what I'm talking about [this](https://gist.github.com/ageis/f5595e59b1cddb1513d1b425a323db04) is a reasonable starting point.

Reply:
Thanks. I've applied the suggested starting point from that article